src/client/app/redux/api/authApi.ts

```diff
@@ -0,0 +1,60 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { currentUserSlice } from '../slices/currentUserSlice';
+import { User } from '../../types/items';
+import { deleteToken, getToken, hasToken } from '../../utils/token';
+import { baseApi } from './baseApi';
+
+type LoginResponse = User & {
+	token: string
+};
+
+export const authApi = baseApi.injectEndpoints({
+	endpoints: builder => ({
+		login: builder.mutation<LoginResponse, { email: string, password: string }>({
+			query: loginArgs => ({
+				url: 'api/login',
+				method: 'POST',
+				body: loginArgs
+			}),
+			// When this mutation is successful, cache stores with these tags are marked as invalid
+			// next time the corresponding endpoint is queried, cache will be ignored and overwritten by a fresh query.
+			// in this case, a user logged in which means that some info for ADMIN meters groups etc.
+			// invalidate forces a refetch to any subscribed components or the next query.
+			invalidatesTags: ['MeterData', 'GroupData']
+			// Listeners for this query (ExtraReducers):
```

Comment:
What is this commented out code for?

Reply:
RTK Slices have an optional property called extra reducers, which can listen to any action defined outside of itself. Sometimes its difficult to know which slices are listening to things like rtkQuery actions. In this case its current user slice listens for the match fulfilled action. This was an early on idea- that I'm not certain this will work cleanly. It wasto document the listeners for the endpoint, but should probably go.

Reply:
removing for now since not standard.